contracts/InterchainTokenService.sol

```diff
@@ -44,7 +44,7 @@ contract InterchainTokenService is
     using AddressBytes for bytes;
     using AddressBytes for address;
 
-    IAxelarGateway public immutable gateway;
+    IAxelarGMPGatewayWithToken public immutable gatewayAddress;
```

Comment:
```suggestion
    IAxelarGMPGateway public immutable gateway;
```
I'd rather keep this pure GMP, and explicitly cast to GMP with token when needed, since that's legacy functionality

Add comments around pure gmp amplifier vs gmp with token legacy gateway support for ITS (latter is expected to revert and only work on chains using Axelar consensus connection)

Reply:
I see. Updated.